x-pack/plugins/asset_manager/server/plugin.ts

```diff
@@ -15,26 +15,48 @@ import {
   PluginConfigDescriptor,
   Logger,
 } from '@kbn/core/server';
+
 import { upsertTemplate } from './lib/manage_index_templates';
+import { startImplicitCollection } from './lib/implicit_collection';
 import { setupRoutes } from './routes';
 import { assetsIndexTemplateConfig } from './templates/assets_template';
 
 export type AssetManagerServerPluginSetup = ReturnType<AssetManagerServerPlugin['setup']>;
-export interface AssetManagerConfig {
-  alphaEnabled?: boolean;
-}
+
+const configSchema = schema.object({
+  alphaEnabled: schema.maybe(schema.boolean()),
+  implicitCollection: schema.maybe(
+    schema.object({
+      enabled: schema.maybe(schema.boolean({ defaultValue: true })),
+      interval: schema.duration({ defaultValue: '5m' }),
+      input: schema.object({
+        hosts: schema.string(),
+        username: schema.string(),
+        password: schema.string(),
+      }),
+      output: schema.object({
+        hosts: schema.string(),
+        username: schema.string(),
+        password: schema.string(),
+      }),
```

Comment:
Using the default client with `kibana_system` user errors out when writing documents and we can't pass `elastic` as user anymore so I ended adding the output as a workaround

```
action [indices:data/write/bulk[s]] is unauthorized for user [kibana_system] with effective roles [kibana_system] on indices [assets-container-default], this action is granted by the index privileges [create_doc,create,delete,index,write,all]
```

Reply:
One way around this is to create a "write to assets" role and add it to the kibana_system user in the write-to cluster. We have to create that role for the output config anyway, so if we add it to kibana_system instead, it should work without the output config. I tested this by using the same config values in output as I use in the top level elasticsearch configs, and it seems to work.

<img width="1372" alt="Screenshot 2023-05-10 at 1 36 01 PM" src="https://github.com/elastic/kibana/assets/159370/2aa1d3cb-c62c-4528-a7fb-adc21efff3c1">

<img width="1710" alt="Screenshot 2023-05-10 at 1 36 18 PM" src="https://github.com/elastic/kibana/assets/159370/4bdc701b-431f-4cae-a993-604626ccb602">

I don't even know if the role needs "manage index templates" since I think it uses the main client for bootstrapping the index template? But maybe something like this is the easiest path?

Reply:
Using `elastic` user in output config works ootb to avoid the privileges setup in local env. I'll make the output config optional and fall back to the default client if not set, and we can rely on the logs to surface lack of privileges requiring additional setup. We can also document this requirement in dev docs.

Reply:
Oh I misunderstood, I thought there was some reason we couldn't use that user anymore? What was the initial problem?

Reply:
We can't use `elastic` user in `elasticsearch.username` anymore, so we can't instantiate the default client with these credentials and thus can't use it in implicit collection. The `output` configuration bypass that validation by using raw `schema.string()` so we can actually use it there as a workaround

> Error: [config validation of [elasticsearch].username]: value of "elastic" is forbidden. This is a superuser account that cannot write to system indices that Kibana needs to function. Use a service account token instead

Reply:
Made input and output elasticsearch optional with fallback to the default client